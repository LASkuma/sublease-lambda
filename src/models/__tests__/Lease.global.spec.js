const LeaseFactory = require('../Lease').default;
const { Lease } = require('../Lease');

describe('Lease Model', () => {
  it('should fail Validation', async () => {
    expect.assertions(1);

    expect(LeaseFactory.create({})).rejects.toThrowErrorMatchingSnapshot();
  });

  it('should return Lease object when validation is passed', async () => {
    expect.assertions(2);

    const result = await LeaseFactory.create({
      address: '808 Columbus Ave',
      state: 'New York',
      city: 'New York',
      description: 'Very nice room',
      price: 1300,
      from: new Date('2018-05-01'),
      bedrooms: 3,
      bathrooms: 1,
      type: 'Bedroom',
      pictureId: '1',
      pictureNumber: 5,
    });

    expect(result).toBeInstanceOf(Lease);
    expect(result).toMatchSnapshot();
  });
});
