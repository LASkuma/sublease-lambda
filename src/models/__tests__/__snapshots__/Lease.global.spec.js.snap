// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lease Model should fail Validation 1`] = `"child \\"address\\" fails because [\\"address\\" is required]"`;

exports[`Lease Model should return Lease object when validation is passed 1`] = `
Lease {
  "address": "808 Columbus Ave",
  "bathrooms": 1,
  "bedrooms": 3,
  "city": "New York",
  "description": "Very nice room",
  "from": 2018-05-01T00:00:00.000Z,
  "price": 1300,
  "state": "New York",
  "type": "Bedroom",
}
`;
